roles/openshift_hosted/tasks/router/router.yml

```diff
@@ -20,12 +20,39 @@
   - "{{ (openshift_hosted_router_certificate | default({'cafile':none})).cafile }}"
   when: openshift_hosted_router_certificate is defined
 
+- name: Generate wildcard router certificate
+  command: >
+    {{ openshift.common.client_binary }} adm ca create-server-cert
+    --config={{ openshift_hosted_kubeconfig }}
+    --signer-cert={{ openshift_master_config_dir }}/ca.crt
+    --signer-key={{ openshift_master_config_dir }}/ca.key
+    --signer-serial={{ openshift_master_config_dir }}/ca.serial.txt
+    --hostnames='*.{{ openshift_master_default_subdomain }}'
+    --cert='{{ openshift_master_config_dir }}/openshift-router.crt'
+    --key='{{ openshift_master_config_dir }}/openshift-router.key'
+  when: (openshift_hosted_router_create_certificate | default(false)) and (not '{{ openshift_master_config_dir }}/openshift-router.key' | exists)
```

Comment:
You should be able to use the module being added in this PR: https://github.com/openshift/openshift-ansible/pull/3378